persistence/src/test/java/cz/muni/fi/pa165/HouseDaoTest.java

```diff
@@ -31,68 +39,91 @@
     public EntityManager em;
 
     @Autowired
-    public HouseDao houseDao;
+    private HouseDao houseDao;
 
-    private House h1;
-    private House h2;
+    private House house;
 
 
     @BeforeMethod
     public void createHouses() {
+        house = new House()
+                .setName("SCALA")
+                .setAddress("MORAVAK")
+                .setHistory("Brief history of time")
+                .setHauntingFrom(Calendar.getInstance().getTime());
 
-        h1 = new House();
-        h2 = new House();
-        h1.setName("SCALA");
-        h2.setName("MORGAL");
-
+        houseDao.create(house);
 
     }
+
     @Test
     public void createHouseTest() {
+        House h1 = new House()
+                .setName("House")
+                .setAddress("Adress")
+                .setHistory("Brief history of time")
+                .setHauntingFrom(Calendar.getInstance().getTime());
+
         assertThat(h1.getId()).isNull();
         houseDao.create(h1);
         assertThat(h1.getId()).isNotNull();
+
     }
 
     @Test
-    public void deleteHouseTest() {
+    public void updateTest(){
+        House h1 = new House()
+                .setName("TRUBA")
+                .setAddress("stramberk")
+                .setHistory("Brief history of time")
+                .setHauntingFrom(Calendar.getInstance().getTime());
         houseDao.create(h1);
-        houseDao.delete(h1);
-
-        assertThat(houseDao.getByName("SCALA")).isNull();
+        h1.setAddress("STRAMBERK");
+        houseDao.update(h1);
+        assertThat(houseDao.getById(h1.getId())).hasFieldOrPropertyWithValue("address", "STRAMBERK");
     }
 
     @Test
-    public void getByNameTest() {
+    public void deleteTest(){
+        House h1 = new House()
+                .setName("PtaciHnizdo")
+                .setAddress("Babisov")
+                .setHistory("Brief history of time")
+                .setHauntingFrom(Calendar.getInstance().getTime());
+
+        assertThat(h1.getId()).isNull();
         houseDao.create(h1);
-        assertThat(houseDao.getByName("SCALA")).isEqualTo(h1);
+        assertThat(h1.getId()).isNotNull();
+        houseDao.delete(h1);
+        assertThat(houseDao.getById(h1.getId())).isNull();
```

Comment:
to to je ok, az na to ze to nije jednoduchy test, tu testujes i create i delete, treba to okresat